src/VisualStudio/Core/Def/Implementation/Progression/GraphQueries/ContainsChildrenGraphQuery.cs

```diff
@@ -40,26 +40,34 @@ internal sealed class ContainsChildrenGraphQuery : IGraphQuery
-
-                                // Todo: we may need fallback to check if this node actually represents a C# or VB language 
-                                // even when its extension fails to say so. One option would be to call DTEWrapper.IsRegisteredForLangService,
-                                // which may not be called here however since deadlock could happen.
-
                                 // The Uri returned by `GetNestedValueByName()` above isn't necessarily absolute and the `OriginalString` is
                                 // the only property that doesn't throw if the UriKind is relative, so `OriginalString` must be used instead
                                 // of `AbsolutePath`.
-                                string ext = Path.GetExtension(uri.OriginalString);
-                                if (ext.Equals(".cs", StringComparison.OrdinalIgnoreCase) || ext.Equals(".vb", StringComparison.OrdinalIgnoreCase))
+                                var path = uri.OriginalString;
+
+                                // We have seen cases where the URI in the graph node has the following
+                                // form, including the quotes (which are invalid path characters):
+                                //     C:\path\to\"some path\App.config"
+                                // So, we choose to be defensive in the presence of invalid path characters.
+                                if (path.IndexOfAny(Path.GetInvalidPathChars()) == -1)
                                 {
-                                    graphBuilder.AddDeferredPropertySet(node, DgmlNodeProperties.ContainsChildren, false);
+                                    // Since a solution load is not yet completed, there is no document available to answer this query.
+                                    // The solution explorer presumes that if somebody doesn't answer for a file, they never will. 
+                                    // See Providers\GraphContextAttachedCollectionSource.cs for more. Therefore we should answer by setting
+                                    // ContainsChildren property to either true or false, so any following updates will be tractable.
+                                    // We will set it to false since the solution explorer assumes the default for this query response is 'false'.
+
+                                    // Todo: we may need fallback to check if this node actually represents a C# or VB language 
+                                    // even when its extension fails to say so. One option would be to call DTEWrapper.IsRegisteredForLangService,
+                                    // which may not be called here however since deadlock could happen.
+
+                                    string ext = Path.GetExtension(path);
+                                    if (ext.Equals(".cs", StringComparison.OrdinalIgnoreCase) || ext.Equals(".vb", StringComparison.OrdinalIgnoreCase))
```

Comment:
Should we just dispense of the Path checks entirely and do path.EndsWith(".cs") || path.EndsWith(".vb")? Not sure why we didn't do that originally?

Reply:
Maybe `Path.GetExtension` just felt more official. 😃 I don't care much either way.

Reply:
`Path.GetExtension` calls `CheckInvalidPathChars(path)`, so it's O(n) instead of O(1). If this gets called frequently or if the input is expected to have invalid path characters, `EndsWith` may be preferable.

Reply:
Well, it might be more official. It's also breaking you. So...